Shorewall(6): add full packages.

wvdakker opened the pull request:
Maintainer: wvdakker@wilsoft.nl

Thanks to lucize for testing it on futro s550 systems with SD cards.

Description:
	Shorewall allows building IPv4/6 firewalls. 
	This is the full Shorewall product which will compile Shorewall scripts.
	It is not recommended to run it on a low memory system.

       This full Shorewal packages also installs Perl which can make the image big (about +2M).

pprindeville commented on net/shorewall/Makefile:
```diff
@@ -0,0 +1,118 @@
+	$(INSTALL_DIR) $(1)/usr/share/shorewall
+	$(INSTALL_DIR) $(1)/usr/lib/shorewall
+	$(CP) $(PKG_INSTALL_DIR)/etc/init.d/shorewall $(1)/etc/init.d/
+	$(CP) $(PKG_INSTALL_DIR)/etc/shorewall/* $(1)/etc/shorewall
```
Any reason to not use:

`$(CP) src/. dst/`

instead?  Are there dot-files that you want to omit (since they won't match the '*' glob)?

wvdakker replied:
Done

pprindeville commented on net/shorewall/files/hostname:
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+cat /proc/sys/kernel/hostname
```
Is there a reason to not use the `hostname` command here?

wvdakker replied:
Because hostname is not installed by default.

stintel replied:
What about ``uci get system.@system[0].hostname``

pprindeville commented on net/shorewall/files/hotplug_iface:
```diff
@@ -0,0 +1,12 @@
+
+# should restart shorewall when an interface comes up
+
+[ ifup = "$ACTION" ] && {
```
Any reason to not use a `case`?

wvdakker replied:
Changed to case

pprindeville commented on net/shorewall/files/vardir:
```diff
@@ -0,0 +1,3 @@
+# move state dir out of ram
```
Don't understand this comment... `/tmp` is a ramfs on OpenWRT.

wvdakker replied:
Removed

pprindeville commented on net/shorewall/patches/120-LOGFILE.patch:
```diff
@@ -0,0 +1,13 @@
+ ###############################################################################
+ 
+-LOG_LEVEL="info"
++LOG_LEVEL="alert"
```
`alert` seems excessively high.  What about `warn`?

wvdakker replied:
changed.

pprindeville commented on net/shorewall6/Makefile:
```diff
@@ -0,0 +1,114 @@
+#
```
I don't think you need two different projects.  See how `isc-dhcp` handles building `dhcpd` and `dhcp6d` from the same `Makefile`.

wvdakker replied:
Upstream has split shorewall in seperate source-packages. IPv6 is an different tar.gz then IPv4.
So combining it into one package is not logical.

wvdakker replied:
Hello Philip,

Can you review #4486 again?
Thanks

/Willem

stintel replied:
Different package, separate commit. You get it by now ;-)

pprindeville replied:
> Upstream has split shorewall in seperate source-packages. IPv6 is an different tar.gz then IPv4.
> So combining it into one package is not logical.

For what it’s worth, packages like usb-utils is built from 2 tarballs and so are some others.

That said, someone might want IPv4 but not v6 or vice versa.

pprindeville commented on net/shorewall6/Makefile:
```diff
@@ -0,0 +1,114 @@
+	$(INSTALL_DIR) $(1)/etc/hotplug.d/iface
+	$(INSTALL_DIR) $(1)/usr/sbin
+	$(INSTALL_DIR) $(1)/usr/share/shorewall6
+	$(CP) $(PKG_INSTALL_DIR)/etc/init.d/shorewall6 $(1)/etc/init.d
```
Please terminate directory names with slash.

wvdakker replied:
Done

pprindeville commented on net/shorewall6/Makefile:
```diff
@@ -0,0 +1,114 @@
+	$(CP) $(PKG_INSTALL_DIR)/etc/init.d/shorewall6 $(1)/etc/init.d
+	$(CP) $(PKG_INSTALL_DIR)/etc/shorewall6/* $(1)/etc/shorewall6
+	$(INSTALL_BIN) ./files/hotplug_iface $(1)/etc/hotplug.d/iface/05-shorewall6
+	$(INSTALL_BIN) ./files/vardir $(1)/etc/shorewall6/vardir
```
If the destination filename is staying the same, you should omit it, i.e.

`$(INSTALL_BIN) ./files/hostname $(1)/etc/shorewall6/`

wvdakker replied:
Done

pprindeville commented on net/shorewall6/files/hotplug_iface:
```diff
@@ -0,0 +1,12 @@
+
+# should restart shorewall6 when an interface comes up
+
+[ ifup = "$ACTION" ] && {
```
Use a `case`.

wvdakker replied:
Changed to case

pprindeville commented on net/shorewall6/files/vardir:
```diff
@@ -0,0 +1,3 @@
+# move state dir out of ram
```
Per above...

wvdakker replied:
Removed

pprindeville commented on net/shorewall6/patches/120-LOGFILE.patch:
```diff
@@ -0,0 +1,13 @@
+ ###############################################################################
+ 
+-LOG_LEVEL="info"
++LOG_LEVEL="alert"
```
Per above.

wvdakker replied:
Changed

lucize commented on net/shorewall/files/vardir:
```diff
@@ -0,0 +1,3 @@
+# move state dir out of ram
```
it's a typo, file got copied from shorewall-lite

lucize commented on net/shorewall/patches/020-init_sh.patch:
```diff
@@ -0,0 +1,97 @@
++
++#
++# The installer may alter this
++#
```
please re add
. /usr/share/shorewall/shorewallrc

wvdakker replied:
Corrected.

pprindeville replied:
Should this be a patch or just a file copied out of `files/etc/` or wherever?  Also, why do we need this in addition to the other startup scripts?  Is this for the non-procd cases?

pprindeville commented:
Any reason to not use procd for the startup?

lucize commented:
@wvdakker can you please address the issues raised or should I add a new PR continuing your work ?
I would really like to have this in tree because mwan3 is not working as expected in my scenarios and adding lots of manual iptables rules is not so cool

wvdakker commented:
Hi,

I had a busy period but will continue working on this PR this week.

/Willem

On Mon, 2017-10-02 at 11:49 -0700, Lucian CRISTIAN wrote:
> @wvdakker can you please address the issues raised or should I add a
> new PR continuing your work ?
> 
> I would really like to have this in tree because mwan3 is not working
> as expected in my scenarios and adding lots of manual iptables rules
> is not so cool
> 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 
> 
> 
>   
>   
> 
> 
> 
> 
> 
> {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c5
> 5493e4bb","name":"GitHub"},"entity":{"external_key":"github/openwrt/p
> ackages","title":"openwrt/packages","subtitle":"GitHub
> repository","main_image_url":"https://cloud.githubusercontent.com/ass
> ets/143418/17495839/a5054eac-5d88-11e6-95fc-
> 7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent
> .com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-
> b52498112777.png","action":{"name":"Open in GitHub","url":"https://gi
> thub.com/openwrt/packages"}},"updates":{"snippets":[{"icon":"PERSON",
> "message":"@lucize in #4486: @wvdakker can you please address the
> issues raised or should I add a new PR continuing your work ?\r\nI
> would really like to have this in tree because mwan3 is not working
> as expected in my scenarios and adding lots of manual iptables rules
> is not so cool"}],"action":{"name":"View Pull Request","url":"https:/
> /github.com/openwrt/packages/pull/4486#issuecomment-333628548"}}}

wvdakker commented:
I will migrate it to procd after the PR is merged. So the *-lite version will also be migrated then.

wvdakker commented:
Lucize can you test the new upload?

Thanks.

wvdakker commented:
Can you test the new upload?

/Willem

On Mon, 2017-10-02 at 11:49 -0700, Lucian CRISTIAN wrote:
> @wvdakker can you please address the issues raised or should I add a
> new PR continuing your work ?
> 
> I would really like to have this in tree because mwan3 is not working
> as expected in my scenarios and adding lots of manual iptables rules
> is not so cool
> 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 
> 
> 
>   
>   
> 
> 
> 
> 
> 
> {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c5
> 5493e4bb","name":"GitHub"},"entity":{"external_key":"github/openwrt/p
> ackages","title":"openwrt/packages","subtitle":"GitHub
> repository","main_image_url":"https://cloud.githubusercontent.com/ass
> ets/143418/17495839/a5054eac-5d88-11e6-95fc-
> 7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent
> .com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-
> b52498112777.png","action":{"name":"Open in GitHub","url":"https://gi
> thub.com/openwrt/packages"}},"updates":{"snippets":[{"icon":"PERSON",
> "message":"@lucize in #4486: @wvdakker can you please address the
> issues raised or should I add a new PR continuing your work ?\r\nI
> would really like to have this in tree because mwan3 is not working
> as expected in my scenarios and adding lots of manual iptables rules
> is not so cool"}],"action":{"name":"View Pull Request","url":"https:/
> /github.com/openwrt/packages/pull/4486#issuecomment-333628548"}}}

lucize commented:
yes, but later or tomorrow

lucize commented on net/shorewall/Makefile:
```diff
@@ -0,0 +1,118 @@
+
+PKG_SOURCE_URL:=http://www.shorewall.net/pub/shorewall/$(PKG_DIRECTORY)/shorewall-$(PKG_MAINVERSION)/ 
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
+PKG_HASH:=4650e2ec73dcf6c38cf2992fefdbd2e302a34a9687a933d8a375d227df2621db96c9a2e54ebbfac50014152caffc0736a7bc1192f479c69c56f78f7874bc09b3
```
typo ? sha256sum c561dd91c4b88ef087a6d172bafb4ac5d01c0055c47b855568536a0c422ff232
on 6 the same

wvdakker replied:
Dont understand what you mean. Hashes are different.

Or did you mean PKG_HASH must be SHA256sum?

lucize replied:
on me is failing with no md5sum or sha256sum format type

stintel replied:
IT has to be PKG_HASH and we only accept sha256 hashes.

lucize commented on net/shorewall/Makefile:
```diff
@@ -0,0 +1,118 @@
+PKG_VERSION:=5.1.4.1
+PKG_DIRECTORY:=5.1
+PKG_MAINVERSION:=5.1.4
+PKG_RELEASE:=2
```
no need to bump the release as is not in tree at the moment

wvdakker replied:
It depends who is reviewing ;)

stintel replied:
New package -> PKG_RELEASE:=1

wvdakker commented:
Are you building on OpenWrt or Lede?
Afaik PKG_HASH is new for Lede.
https://lists.openwrt.org/pipermail/openwrt-devel/2017-January/042537.h
tml id="-x-evo-selection-start-marker">

On Wed, 2017-10-04 at 06:56 +0000, Lucian CRISTIAN wrote:
> @lucize commented on this pull request.
> 
> 
> 
> 
> In net/shorewall/Makefile:
> 
> > +#
> +# This is free software, licensed under the GNU General Public
> License v2.
> +# See /LICENSE for more information.
> +#
> +
> +include $(TOPDIR)/rules.mk
> +
> +PKG_NAME:=shorewall
> +PKG_VERSION:=5.1.4.1
> +PKG_DIRECTORY:=5.1
> +PKG_MAINVERSION:=5.1.4
> +PKG_RELEASE:=2
> +
> +PKG_SOURCE_URL:=http://www.shorewall.net/pub/shorewall/$(PKG_DIRECTO
> RY)/shorewall-$(PKG_MAINVERSION)/ 
> +PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
> +PKG_HASH:=4650e2ec73dcf6c38cf2992fefdbd2e302a34a9687a933d8a375d227df
> 2621db96c9a2e54ebbfac50014152caffc0736a7bc1192f479c69c56f78f7874bc09b
> 3
> 
> on me is failing with no md5sum or sha256sum format type
> 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 
> 
> 
>   
>   
> 
> 
> 
> 
> 
> {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c5
> 5493e4bb","name":"GitHub"},"entity":{"external_key":"github/openwrt/p
> ackages","title":"openwrt/packages","subtitle":"GitHub
> repository","main_image_url":"https://cloud.githubusercontent.com/ass
> ets/143418/17495839/a5054eac-5d88-11e6-95fc-
> 7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent
> .com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-
> b52498112777.png","action":{"name":"Open in GitHub","url":"https://gi
> thub.com/openwrt/packages"}},"updates":{"snippets":[{"icon":"PERSON",
> "message":"@lucize commented on #4486"}],"action":{"name":"View Pull
> Request","url":"https://github.com/openwrt/packages/pull/4486#discuss
> ion_r142592945"}}}

lucize commented:
for LEDE, latest tree, it was the example of sha256sum for
c561dd91c4b88ef087a6d172bafb4ac5d01c0055c47b855568536a0c422ff232  shorewall-5.1.4.1.tar.bz2
and for 6 is
e3da805b4aa1e8c483f1bb451cb47a869e1977b213a86145a82d42e9bc12ea36  shorewall6-5.1.4.1.tar.bz2
on core and lite version you have them correctly introduced.
Here, you can see that they are too long

wvdakker commented:
I am confused here :)
the SHA512 checksum of shorewall (4), http://www.shorewall.net/pub/shor
ewall/5.1/shorewall-5.1.4/5.1.4.1.sha512sums, is
4650e2ec73dcf6c38cf2992fefdbd2e302a34a9687a933d8a375d227df2621db96c9a2e
54ebbfac50014152caffc0736a7bc1192f479c69c56f78f7874bc09b3
which gave me no problems in compiling and installing.

On Wed, 2017-10-04 at 07:19 +0000, Lucian CRISTIAN wrote:
> for LEDE, latest tree, it was the example of sha256sum for
> 
> c561dd91c4b88ef087a6d172bafb4ac5d01c0055c47b855568536a0c422ff232  sho
> rewall-5.1.4.1.tar.bz2
> 
> and for 6 is
> 
> e3da805b4aa1e8c483f1bb451cb47a869e1977b213a86145a82d42e9bc12ea36  sho
> rewall6-5.1.4.1.tar.bz2
> 
> on core and lite version you have them correctly introduced.
> 
> Here, you can see that they are too long
> 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 
> 
> 
>   
>   
> 
> 
> 
> 
> 
> {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c5
> 5493e4bb","name":"GitHub"},"entity":{"external_key":"github/openwrt/p
> ackages","title":"openwrt/packages","subtitle":"GitHub
> repository","main_image_url":"https://cloud.githubusercontent.com/ass
> ets/143418/17495839/a5054eac-5d88-11e6-95fc-
> 7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent
> .com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-
> b52498112777.png","action":{"name":"Open in GitHub","url":"https://gi
> thub.com/openwrt/packages"}},"updates":{"snippets":[{"icon":"PERSON",
> "message":"@lucize in #4486: for LEDE, latest tree, it was the
> example of sha256sum
> for\r\nc561dd91c4b88ef087a6d172bafb4ac5d01c0055c47b855568536a0c422ff2
> 32  shorewall-5.1.4.1.tar.bz2\r\nand for 6
> is\r\ne3da805b4aa1e8c483f1bb451cb47a869e1977b213a86145a82d42e9bc12ea3
> 6  shorewall6-5.1.4.1.tar.bz2\r\non core and lite version you have
> them correctly introduced.\r\nHere, you can see that they are too
> long"}],"action":{"name":"View Pull Request","url":"https://github.co
> m/openwrt/packages/pull/4486#issuecomment-334070281"}}}

lucize commented:
if you have it  already downloaded is no problem, didn't see an announcement that sha512sum is the new default hash
on clean build I'm having this:

```
build@build:~/spital_sus$ make package/shorewall/prepare V=s
make[1]: Entering directory '/home/build/spital_sus'
make[2]: Entering directory '/home/build/spital_sus/feeds/packages/net/shorewall'
mkdir -p /home/build/spital_sus/dl
SHELL= flock /home/build/spital_sus/tmp/.shorewall-5.1.4.1.tar.bz2.flock -c '   /home/build/spital_sus/scripts/download.pl "/home/build/spital_sus/dl" "shorewall-5.1.4.1.tar.bz2" "4650e2ec73dcf6c38cf2992fefdbd2e302a34a9687a933d8a375d227df2621db96c9a2e54ebbfac50014152caffc0736a7bc1192f479c69c56f78f7874bc09b3" "" "http://www.shorewall.net/pub/shorewall/5.1/shorewall-5.1.4/"    '
Cannot find appropriate hash command, ensure the provided hash is either a MD5 or SHA256 checksum.
Makefile:118: recipe for target '/home/build/spital_sus/dl/shorewall-5.1.4.1.tar.bz2' failed
make[2]: *** [/home/build/spital_sus/dl/shorewall-5.1.4.1.tar.bz2] Error 255
make[2]: Leaving directory '/home/build/spital_sus/feeds/packages/net/shorewall'
package/Makefile:109: recipe for target 'package/feeds/packages/shorewall/prepare' failed
make[1]: *** [package/feeds/packages/shorewall/prepare] Error 2
make[1]: Leaving directory '/home/build/spital_sus'
/home/build/spital_sus/include/toplevel.mk:207: recipe for target 'package/shorewall/prepare' failed

```

wvdakker commented:
Output after removing source, clean:

make[1]: Entering directory '/home/test/Develop/Lede/source'
make[2]: Entering directory
'/home/test/Develop/Lede/packages/net/shorewall'
mkdir -p /home/test/Develop/Lede/source/dl
SHELL= flock /home/test/Develop/Lede/source/tmp/.shorewall-
5.1.4.1.tar.bz2.flock -c '  	/home/test/Develop/Lede/source/scri
pts/download.pl "/home/test/Develop/Lede/source/dl" "shorewall-
5.1.4.1.tar.bz2"
"4650e2ec73dcf6c38cf2992fefdbd2e302a34a9687a933d8a375d227df2621db96c9a2
e54ebbfac50014152caffc0736a7bc1192f479c69c56f78f7874bc09b3" "" "http://
www.shorewall.net/pub/shorewall/5.1/shorewall-5.1.4/"    '
  % Total    % Received % Xferd  Average
Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left 
 Speed
100  498k  100  498k    0     0   249k      0  0:00:02  0:00:02 --:--:-
-  226k
. /home/test/Develop/Lede/source/include/shell.sh; bzcat
/home/test/Develop/Lede/source/dl/shorewall-5.1.4.1.tar.bz2 | tar -C
/home/test/Develop/Lede/source/build_dir/target-mips_24kc_musl-
1.1.16/shorewall-5.1.4.1/.. -xf - 
[ ! -d ./src/ ] || cp -fpR ./src/*
/home/test/Develop/Lede/source/build_dir/target-mips_24kc_musl-
1.1.16/shorewall-5.1.4.1

Applying ./patches/010-update_install_sh.patch using plaintext: 
patching file install.sh

Applying ./patches/020-init_sh.patch using plaintext: 
patching file init.openwrt.sh

Applying ./patches/120-LOGFILE.patch using plaintext: 
patching file configfiles/shorewall.conf
touch /home/test/Develop/Lede/source/build_dir/target-mips_24kc_musl-
1.1.16/shorewall-5.1.4.1/.prepared_c880b2d7647b74f9440b7b7270b8c712
make[2]: Leaving directory
'/home/test/Develop/Lede/packages/net/shorewall'
make[1]: Leaving directory '/home/test/Develop/Lede/source'


On Wed, 2017-10-04 at 08:36 +0000, Lucian CRISTIAN wrote:
> if you have it  already downloaded is no problem, didn't see an
> announcement that sha512sum is the new default hash
> 
> on clean build I'm having this:
> 
> build@build:~/spital_sus$ make package/shorewall/prepare V=s
> make[1]: Entering directory '/home/build/spital_sus'
> make[2]: Entering directory
> '/home/build/spital_sus/feeds/packages/net/shorewall'
> mkdir -p /home/build/spital_sus/dl
> SHELL= flock /home/build/spital_sus/tmp/.shorewall-
> 5.1.4.1.tar.bz2.flock -c
> '   /home/build/spital_sus/scripts/download.pl
> "/home/build/spital_sus/dl" "shorewall-5.1.4.1.tar.bz2"
> "4650e2ec73dcf6c38cf2992fefdbd2e302a34a9687a933d8a375d227df2621db96c9
> a2e54ebbfac50014152caffc0736a7bc1192f479c69c56f78f7874bc09b3" "" "htt
> p://www.shorewall.net/pub/shorewall/5.1/shorewall-5.1.4/"    '
> Cannot find appropriate hash command, ensure the provided hash is
> either a MD5 or SHA256 checksum.
> Makefile:118: recipe for target '/home/build/spital_sus/dl/shorewall-
> 5.1.4.1.tar.bz2' failed
> make[2]: *** [/home/build/spital_sus/dl/shorewall-5.1.4.1.tar.bz2]
> Error 255
> make[2]: Leaving directory
> '/home/build/spital_sus/feeds/packages/net/shorewall'
> package/Makefile:109: recipe for target
> 'package/feeds/packages/shorewall/prepare' failed
> make[1]: *** [package/feeds/packages/shorewall/prepare] Error 2
> make[1]: Leaving directory '/home/build/spital_sus'
> /home/build/spital_sus/include/toplevel.mk:207: recipe for target
> 'package/shorewall/prepare' failed
> 
> 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 
> 
> 
>   
>   
> 
> 
> 
> 
> 
> {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c5
> 5493e4bb","name":"GitHub"},"entity":{"external_key":"github/openwrt/p
> ackages","title":"openwrt/packages","subtitle":"GitHub
> repository","main_image_url":"https://cloud.githubusercontent.com/ass
> ets/143418/17495839/a5054eac-5d88-11e6-95fc-
> 7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent
> .com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-
> b52498112777.png","action":{"name":"Open in GitHub","url":"https://gi
> thub.com/openwrt/packages"}},"updates":{"snippets":[{"icon":"PERSON",
> "message":"@lucize in #4486: if you have it  already downloaded is no
> problem, didn't see an announcement that sha512sum is the new default
> hash\r\non clean build I'm having
> this:\r\n\r\n```\r\nbuild@build:~/spital_sus$ make
> package/shorewall/prepare V=s\r\nmake[1]: Entering directory
> '/home/build/spital_sus'\r\nmake[2]: Entering directory
> '/home/build/spital_sus/feeds/packages/net/shorewall'\r\nmkdir -p
> /home/build/spital_sus/dl\r\nSHELL= flock
> /home/build/spital_sus/tmp/.shorewall-5.1.4.1.tar.bz2.flock -c
> '   /home/build/spital_sus/scripts/download.pl
> \"/home/build/spital_sus/dl\" \"shorewall-5.1.4.1.tar.bz2\"
> \"4650e2ec73dcf6c38cf2992fefdbd2e302a34a9687a933d8a375d227df2621db96c
> 9a2e54ebbfac50014152caffc0736a7bc1192f479c69c56f78f7874bc09b3\" \"\"
> \"http://www.shorewall.net/pub/shorewall/5.1/shorewall-5.1.4/\"    '\
> r\nCannot find appropriate hash command, ensure the provided hash is
> either a MD5 or SHA256 checksum.\r\nMakefile:118: recipe for target
> '/home/build/spital_sus/dl/shorewall-5.1.4.1.tar.bz2'
> failed\r\nmake[2]: *** [/home/build/spital_sus/dl/shorewall-
> 5.1.4.1.tar.bz2] Error 255\r\nmake[2]: Leaving directory
> '/home/build/spital_sus/feeds/packages/net/shorewall'\r\npackage/Make
> file:109: recipe for target
> 'package/feeds/packages/shorewall/prepare' failed\r\nmake[1]: ***
> [package/feeds/packages/shorewall/prepare] Error 2\r\nmake[1]:
> Leaving directory
> '/home/build/spital_sus'\r\n/home/build/spital_sus/include/toplevel.m
> k:207: recipe for target 'package/shorewall/prepare'
> failed\r\n\r\n``` "}],"action":{"name":"View Pull Request","url":"htt
> ps://github.com/openwrt/packages/pull/4486#issuecomment-334087231"}}}

lucize commented:
now I'm confused why is working for you :)

```
build@build:~/spital_sus$ ./scripts/getver.sh
r4970-a37655baca
build@build:~/spital_sus$ which sha512sum
/usr/bin/sha512sum

```

wvdakker commented:
./scripts/getver.sh
r3435-65eec8bd5f

I am building it in the stable branche (17.01.2)
But that shouldnt be a problem....

On Wed, 2017-10-04 at 08:49 +0000, Lucian CRISTIAN wrote:
> which sha512sum

lucize commented:
do you have an official source that sha512 is an accepted algorithm, I can't find it?
https://github.com/lede-project/source/commit/7416d2e046b87b262b407f8af70b8dd9b2927c70

wvdakker commented:
Thats the only link I could find. 
Busybox supports also the SHA512 hashes. http://lists.infradead.org/pip
ermail/lede-dev/2017-January/005363.html

I shall ask around on IRC what the policy is for PKG_HASH. 

On Wed, 2017-10-04 at 02:41 -0700, Lucian CRISTIAN wrote:
> do you have an official source that sha512 is an accepted algorithm,
> I can't find it?
> 
> lede-project/source@7416d2e
> 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 
> 
> 
>   
>   
> 
> 
> 
> 
> 
> {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c5
> 5493e4bb","name":"GitHub"},"entity":{"external_key":"github/openwrt/p
> ackages","title":"openwrt/packages","subtitle":"GitHub
> repository","main_image_url":"https://cloud.githubusercontent.com/ass
> ets/143418/17495839/a5054eac-5d88-11e6-95fc-
> 7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent
> .com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-
> b52498112777.png","action":{"name":"Open in GitHub","url":"https://gi
> thub.com/openwrt/packages"}},"updates":{"snippets":[{"icon":"PERSON",
> "message":"@lucize in #4486: do you have an official source that
> sha512 is an accepted algorithm, I can't find it?\r\nhttps://github.c
> om/lede-
> project/source/commit/7416d2e046b87b262b407f8af70b8dd9b2927c70\r\n"}]
> ,"action":{"name":"View Pull Request","url":"https://github.com/openw
> rt/packages/pull/4486#issuecomment-334103308"}}}

wvdakker commented:
Change SHA512 to SHA1 (256 not available).
See if you have more luck this time. I had no problems
compiling/installing it.

On Wed, 2017-10-04 at 02:41 -0700, Lucian CRISTIAN wrote:
> do you have an official source that sha512 is an accepted algorithm,
> I can't find it?
> 
> lede-project/source@7416d2e
> 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 
> 
> 
>   
>   
> 
> 
> 
> 
> 
> {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c5
> 5493e4bb","name":"GitHub"},"entity":{"external_key":"github/openwrt/p
> ackages","title":"openwrt/packages","subtitle":"GitHub
> repository","main_image_url":"https://cloud.githubusercontent.com/ass
> ets/143418/17495839/a5054eac-5d88-11e6-95fc-
> 7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent
> .com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-
> b52498112777.png","action":{"name":"Open in GitHub","url":"https://gi
> thub.com/openwrt/packages"}},"updates":{"snippets":[{"icon":"PERSON",
> "message":"@lucize in #4486: do you have an official source that
> sha512 is an accepted algorithm, I can't find it?\r\nhttps://github.c
> om/lede-
> project/source/commit/7416d2e046b87b262b407f8af70b8dd9b2927c70\r\n"}]
> ,"action":{"name":"View Pull Request","url":"https://github.com/openw
> rt/packages/pull/4486#issuecomment-334103308"}}}

lucize commented:
I gave you the correct sha256sum, the package is tested but I was unclear about the hash
Stintel said they would only accept sha256sum https://github.com/openwrt/packages/pull/4486#discussion_r142629459

wvdakker commented:
Corrected. Had a discussion on IRC. I was in the impression we only took upstream provided hashes. 
So my bad. 

Also corrected the shorewall(6)-lite packages.

lucize commented:
being there, you could update all to 5.1.7.1 :)

wvdakker commented:
This means it is working and operating ok for you?

On Wed, 2017-10-04 at 04:01 -0700, Lucian CRISTIAN wrote:
> being there, you could update all to 5.1.7.1 :)
> 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 
> 
> 
>   
>   
> 
> 
> 
> 
> 
> {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c5
> 5493e4bb","name":"GitHub"},"entity":{"external_key":"github/openwrt/p
> ackages","title":"openwrt/packages","subtitle":"GitHub
> repository","main_image_url":"https://cloud.githubusercontent.com/ass
> ets/143418/17495839/a5054eac-5d88-11e6-95fc-
> 7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent
> .com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-
> b52498112777.png","action":{"name":"Open in GitHub","url":"https://gi
> thub.com/openwrt/packages"}},"updates":{"snippets":[{"icon":"PERSON",
> "message":"@lucize in #4486: being there, you could update all to
> 5.1.7.1 :)"}],"action":{"name":"View Pull Request","url":"https://git
> hub.com/openwrt/packages/pull/4486#issuecomment-334121068"}}}

wvdakker commented:
Perhaps updating in a new PR?
On Wed, 2017-10-04 at 04:01 -0700, Lucian CRISTIAN wrote:
> being there, you could update all to 5.1.7.1 :)
> 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 
> 
> 
>   
>   
> 
> 
> 
> 
> 
> {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c5
> 5493e4bb","name":"GitHub"},"entity":{"external_key":"github/openwrt/p
> ackages","title":"openwrt/packages","subtitle":"GitHub
> repository","main_image_url":"https://cloud.githubusercontent.com/ass
> ets/143418/17495839/a5054eac-5d88-11e6-95fc-
> 7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent
> .com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-
> b52498112777.png","action":{"name":"Open in GitHub","url":"https://gi
> thub.com/openwrt/packages"}},"updates":{"snippets":[{"icon":"PERSON",
> "message":"@lucize in #4486: being there, you could update all to
> 5.1.7.1 :)"}],"action":{"name":"View Pull Request","url":"https://git
> hub.com/openwrt/packages/pull/4486#issuecomment-334121068"}}}

lucize commented:
there are minor adjustments:
the following dependency: `kmod-ipt-hashlimit`, `kmod-ipt-raw`, `kmod-ipt-raw6` and `iptables-mod-hashlimit`
```
ERROR: Per-ip log rate limiting requires Hashlimit Match in your kernel and iptables
ERROR: UNTRACKED state requires Raw Table in your kernel and iptables
```

/etc/init.d/shorewall start
`   ERROR: Shorewall Lite does not appear to be installed on this system`
this is odd because function  `set_default_product` returns Shorewall, not Shorewall Lite

manual startup, `shorewall start` is working so because of hotplug `shorewall restart` the system is starting correctly 

I think some regression on `shorewall restart`

```
find: unrecognized: -newer
BusyBox v1.27.2 () multi-call binary.
```
Trying to install full find, with force `opkg install --force-overwrite findutils-find` works
But busybox version is not ok because the edited conf file will not be reloaded, only with a reboot !
This is not happening on an older system BusyBox v1.26.2 

And the last one: `touch: /var/lock/subsys/shorewall: No such file or directory`, maybe changing it to /var/lock or let the user edit the config file

so tu sum up, only iptables mods are blockers and needs to be seen about `findutils-find`

Regards!

wvdakker commented:
> ERROR: Per-ip log rate limiting requires Hashlimit Match in your kernel and iptables
>ERROR: UNTRACKED state requires Raw Table in your kernel and iptables

Added

> ERROR: Shorewall Lite does not appear to be installed on this system

Fixed

> /var/lock/subsys/shorewall: No such file or directory, maybe changing it to /var/lock

Done

lucize commented on net/shorewall/Makefile:
```diff
@@ -0,0 +1,117 @@
+define Package/shorewall
+    SECTION:=net
+    CATEGORY:=Network
+    DEPENDS:=+ip +iptables +kmod-ipt-hashlimit +kmod-ipt-raw +shorewall-core +perl +perlbase-autoloader +perlbase-autouse +perlbase-dynaloader +perlbase-digest +perlbase-findbin +perlbase-getopt +perlbase-hash
```
did you forget about +iptables-mod-hashlimit here ?

wvdakker commented:
Corrected :shame

On Thu, 2017-10-05 at 12:34 +0000, Lucian CRISTIAN wrote:
> @lucize commented on this pull request.
> 
> 
> 
> 
> In net/shorewall/Makefile:
> 
> > +PKG_MAINVERSION:=5.1.4
> +PKG_RELEASE:=2
> +
> +PKG_SOURCE_URL:=http://www.shorewall.net/pub/shorewall/$(PKG_DIRECTO
> RY)/shorewall-$(PKG_MAINVERSION)/ 
> +PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
> +PKG_HASH:=c561dd91c4b88ef087a6d172bafb4ac5d01c0055c47b855568536a0c42
> 2ff232
> +PKG_MAINTAINER:=Willem van den Akker <wvdakker@wilsoft.nl>
> +PKG_LICENSE:=GPL-2.0+
> +PKG_LICENSE_FILES:=COPYING
> +
> +include $(INCLUDE_DIR)/package.mk
> +
> +define Package/shorewall
> +    SECTION:=net
> +    CATEGORY:=Network
> +    DEPENDS:=+ip +iptables +kmod-ipt-hashlimit +kmod-ipt-raw
> +shorewall-core +perl +perlbase-autoloader +perlbase-autouse
> +perlbase-dynaloader +perlbase-digest +perlbase-findbin +perlbase-
> getopt +perlbase-hash
> 
> did you forget about +iptables-mod-hashlimit here ?
> 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 
> 
> 
>   
>   
> 
> 
> 
> 
> 
> {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c5
> 5493e4bb","name":"GitHub"},"entity":{"external_key":"github/openwrt/p
> ackages","title":"openwrt/packages","subtitle":"GitHub
> repository","main_image_url":"https://cloud.githubusercontent.com/ass
> ets/143418/17495839/a5054eac-5d88-11e6-95fc-
> 7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent
> .com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-
> b52498112777.png","action":{"name":"Open in GitHub","url":"https://gi
> thub.com/openwrt/packages"}},"updates":{"snippets":[{"icon":"PERSON",
> "message":"@lucize commented on #4486"}],"action":{"name":"View Pull
> Request","url":"https://github.com/openwrt/packages/pull/4486#pullreq
> uestreview-67357605"}}}

stintel commented on net/shorewall-lite/Makefile:
```diff
@@ -14,7 +14,7 @@ PKG_DIRECTORY:=5.1
 PKG_RELEASE:=1
 
-PKG_SOURCE_URL:=http://www.shorewall.net/pub/shorewall/$(PKG_DIRECTORY)/shorewall-$(PKG_MAINVERSION)/ 
+PKG_SOURCE_URL:=http://www.shorewall.net/pub/shorewall/$(PKG_DIRECTORY)/shorewall-$(PKG_MAINVERSION)/
```
Changes a different package, so should go in a separate commit. Even if it's for the same software.

wvdakker replied:
Done

stintel commented on net/shorewall/Makefile:
```diff
@@ -0,0 +1,117 @@
+/etc/shorewall/tcinterfaces
+/etc/shorewall/tcpri
+/etc/shorewall/tunnels
+/etc/shorewall/zones
```
Why not just /etc/shorewall/ ?

wvdakker replied:
Done

stintel commented on net/shorewall/Makefile:
```diff
@@ -0,0 +1,117 @@
+	$(INSTALL_BIN) ./files/hostname $(1)/etc/shorewall/
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/shorewall/. $(1)/usr/lib/shorewall/
+	$(CP) $(PKG_INSTALL_DIR)/usr/share/shorewall/. $(1)/usr/share/shorewall/
+endef
```
I could nitpick here about alphabetical order, but there are probably many packages that don't do it. Either way I prefer it.

wvdakker replied:
Done

stintel commented on net/shorewall6-lite/Makefile:
```diff
@@ -14,7 +14,7 @@ PKG_DIRECTORY:=5.1
 PKG_RELEASE:=2
 
-PKG_SOURCE_URL:=http://www.shorewall.net/pub/shorewall/$(PKG_DIRECTORY)/shorewall-$(PKG_MAINVERSION)/ 
+PKG_SOURCE_URL:=http://www.shorewall.net/pub/shorewall/$(PKG_DIRECTORY)/shorewall-$(PKG_MAINVERSION)/
```
Same here. Different package -> separate commit.

wvdakker replied:
Now have 2 commits:
- shorewall-core, shorewall-lite, shorewall6-lite
- shorewall, shorewall6

stintel commented on net/shorewall6/Makefile:
```diff
@@ -0,0 +1,113 @@
+/etc/shorewall6/tcpri
+/etc/shorewall6/tunnels
+/etc/shorewall6/zones
+endef
```
Just ``/etc/shorewall6/`` ?

wvdakker replied:
Done

stintel commented on net/shorewall6/files/hostname:
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+cat /proc/sys/kernel/hostname
```
``uci get system.@system[0].hostname`` ?

wvdakker replied:
Done

stintel commented:
On second thought, the init script seems like it could use a rewrite, e.g. it uses SYSCONFDIR which is a RedHat-ism. Are the comments required? Instead of patching the existing init script, I would write a small, clean, ideally procd compatible init script, and add that in the files/ dir instead.

wvdakker commented:
The plan was om after the merge to migrate the init-script to procd.

On Mon, 2017-10-09 at 06:35 -0700, Stijn Tintel wrote:
> On second thought, the init script seems like it could use a rewrite,
> e.g. it uses SYSCONFDIR which is a RedHat-ism. Are the comments
> required? Instead of patching the existing init script, I would write
> a small, clean, ideally procd compatible init script, and add that in
> the files/ dir instead.
> 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 
> 
> 
>   
>   
> 
> 
> 
> 
> 
> {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c5
> 5493e4bb","name":"GitHub"},"entity":{"external_key":"github/openwrt/p
> ackages","title":"openwrt/packages","subtitle":"GitHub
> repository","main_image_url":"https://cloud.githubusercontent.com/ass
> ets/143418/17495839/a5054eac-5d88-11e6-95fc-
> 7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent
> .com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-
> b52498112777.png","action":{"name":"Open in GitHub","url":"https://gi
> thub.com/openwrt/packages"}},"updates":{"snippets":[{"icon":"PERSON",
> "message":"@stintel in #4486: On second thought, the init script
> seems like it could use a rewrite, e.g. it uses SYSCONFDIR which is a
> RedHat-ism. Are the comments required? Instead of patching the
> existing init script, I would write a small, clean, ideally procd
> compatible init script, and add that in the files/ dir
> instead."}],"action":{"name":"View Pull Request","url":"https://githu
> b.com/openwrt/packages/pull/4486#issuecomment-335159848"}}}

wvdakker commented:
@ lucize: can you test the latest packages?
SHorewall is migrated to procd and 5.1.7.2

wvdakker commented:
Hello Stijn,

I made the changes and migrated to procd.
Can you please review it again?

/Willem
On Mon, 2017-10-09 at 06:35 -0700, Stijn Tintel wrote:
> On second thought, the init script seems like it could use a rewrite,
> e.g. it uses SYSCONFDIR which is a RedHat-ism. Are the comments
> required? Instead of patching the existing init script, I would write
> a small, clean, ideally procd compatible init script, and add that in
> the files/ dir instead.
> 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 
> 
> 
>   
>   
> 
> 
> 
> 
> 
> {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c5
> 5493e4bb","name":"GitHub"},"entity":{"external_key":"github/openwrt/p
> ackages","title":"openwrt/packages","subtitle":"GitHub
> repository","main_image_url":"https://cloud.githubusercontent.com/ass
> ets/143418/17495839/a5054eac-5d88-11e6-95fc-
> 7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent
> .com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-
> b52498112777.png","action":{"name":"Open in GitHub","url":"https://gi
> thub.com/openwrt/packages"}},"updates":{"snippets":[{"icon":"PERSON",
> "message":"@stintel in #4486: On second thought, the init script
> seems like it could use a rewrite, e.g. it uses SYSCONFDIR which is a
> RedHat-ism. Are the comments required? Instead of patching the
> existing init script, I would write a small, clean, ideally procd
> compatible init script, and add that in the files/ dir
> instead."}],"action":{"name":"View Pull Request","url":"https://githu
> b.com/openwrt/packages/pull/4486#issuecomment-335159848"}}}

lucize commented:
@wvdakker sorry I'm a bit busy, I'll test it ASAP

lucize commented:
tested and ok (beside the find problem related to busybox) on x86_64 LEDE r5103-f6c4a9c045

diizzyy commented:
@pprindeville @stintel
From what I can tell this seems fine now except that each package directory should have its own commit looking at the comment history?

wvdakker commented:
Split up in 5 commits. Each package in one commit.

/Willem

stintel commented:
Please remove the 'Shorewall-core: ' prefix from the commit messages (not the subject). Also, you mention you changed PKG_HASH from sha512 but this is not what the diff shows, so the message is confusing. I assume this change was made compared to the earlier versions of the PR. I'd just drop it.

wvdakker commented:
commit messages updated.

pprindeville commented on net/shorewall-lite/Makefile:
```diff
@@ -58,19 +57,19 @@ define Build/Compile
-	$(INSTALL_DIR) $(1)/etc/hotplug.d/iface
-	$(INSTALL_DIR) $(1)/etc/shorewall-lite/state
-	$(INSTALL_DIR) $(1)/usr/share
+	$(INSTALL_DIR) $(1)/etc/hotplug.d/iface/
```
Not sure I see the point of changing the order unnecessarily.

wvdakker replied:
One of the reviewers suggested an alphabetic order. Not necessary but would be nice.

pprindeville replied:
I prefer smaller change sets/low touch.

wvdakker replied:
I understand. Is this a showstopper and have I to revert it?
Personally I find an ordered list easier readable.

No problem reverting it if it is a no-go.

pprindeville replied:
Please restore the original ordering (and hence the smallest possible patchest).

Patches with superfluous changes are a lot more work to back out if they ever need to be reverted.

wvdakker replied:
Done.

Will make a PR after this one.

pprindeville commented on net/shorewall-lite/files/shorewall-lite.init:
```diff
@@ -0,0 +1,33 @@
+#!/bin/sh /etc/rc.common
+
+USE_PROCD=1
+
```
Don't need an extra newline here.

wvdakker replied:
Removed newline after USE_PROC

pprindeville commented on net/shorewall-lite/patches/120-logfile.patch:
```diff
@@ -0,0 +1,13 @@
+ ###############################################################################
+ 
+-LOGFILE=
++LOGFILE=/dev/null
```
Why is logging discarded?  Why not use the syslog service?

wvdakker replied:
Default Shorewall logs to syslog and can be read with Logread. The logfile parameter in the configfile points to a separate file which I have disabled by default to spare diskspace.

pprindeville replied:
Okay.

pprindeville commented on net/shorewall/files/shorewall.init:
```diff
@@ -0,0 +1,33 @@
+#!/bin/sh /etc/rc.common
+
+USE_PROCD=1
+
```
Don't need a blank line here.

wvdakker replied:
Done

pprindeville commented on net/shorewall/patches/010-update_install_sh.patch:
```diff
@@ -0,0 +1,23 @@
+     archlinux)
+ 	echo "Installing ArchLinux-specific configuration..."
+ 	;;
++    openwrt)
```
Have you tried to upstream this patch?

wvdakker replied:
No, Will do after the review to avoid mistakes to upstream.

pprindeville commented on net/shorewall6-lite/Makefile:
```diff
@@ -59,19 +58,19 @@ define Build/Compile
-	$(INSTALL_DIR) $(1)/etc/hotplug.d/iface
-	$(INSTALL_DIR) $(1)/etc/shorewall6-lite/state
-	$(INSTALL_DIR) $(1)/usr/share
+	$(INSTALL_DIR) $(1)/etc/hotplug.d/iface/
```
Again, please don't gratuitously change the order of lines.  Go for minimal change sets.

wvdakker replied:
This was a one time change as suggested by a reviewer.

pprindeville commented on net/shorewall6-lite/files/shorewall6-lite.init:
```diff
@@ -0,0 +1,33 @@
+#!/bin/sh /etc/rc.common
+
+USE_PROCD=1
+
```
Don't need a blank line here.

wvdakker replied:
Removed

pprindeville commented on net/shorewall6-lite/patches/120-LOGFILE.patch:
```diff
@@ -1,11 +0,0 @@
- ###############################################################################
- 
--LOGFILE=
-+LOGFILE=/dev/null
```
Ditto about using syslog...

wvdakker replied:
See previous

pprindeville commented on net/shorewall6-lite/patches/120-logfile.patch:
```diff
@@ -0,0 +1,13 @@
+ ###############################################################################
+ 
+-LOGFILE=
++LOGFILE=/dev/null
```
Ditto...

pprindeville replied:
Also, does this need to be a patch?  Why not just drop it into `files/etc/` or whatever?

wvdakker replied:
I decided to make it a patch because of possible changes of upstream in the config and not to check the conf files for internal changes.

pprindeville commented on net/shorewall6/files/shorewall6.init:
```diff
@@ -0,0 +1,33 @@
+#!/bin/sh /etc/rc.common
+
+USE_PROCD=1
+
```
Don't need a blank line here.

Also, really confused... you have the same files both in `files/` and as patches?  Why?

wvdakker replied:
Correct. Forgot to remove the file from patches.

pprindeville commented on net/shorewall6/patches/010-update_install_sh.patch:
```diff
@@ -0,0 +1,23 @@
+     archlinux)
+ 	echo "Installing ArchLinux-specific configuration..."
+ 	;;
++    openwrt)
```
Should be upstreamed.

wvdakker replied:
Will do after review.

pprindeville commented on net/shorewall6/patches/020-init_sh.patch:
```diff
@@ -0,0 +1,100 @@
+===================================================================
+--- /dev/null	1970-01-01 00:00:00.000000000 +0000
++++ shorewall6-5.1.4.1/init.openwrt.sh	2017-06-06 22:52:41.876937057 +0200
+@@ -0,0 +1,95 @@
```
Put this in `files/`

wvdakker replied:
can be removed because files contains a new init file.
This one was left behind.

pprindeville commented:
Please don’t do Yoda code, i.e.:

if [ ifup = “$ACTION” ] then

pprindeville commented on net/shorewall/Makefile:
```diff
@@ -0,0 +1,73 @@
+define Package/shorewall
+    SECTION:=devel
+    CATEGORY:=Network
+    DEPENDS:=+ip +iptables +kmod-ipt-hashlimit +kmod-ipt-raw +iptables-mod-hashlimit +shorewall-core +perl +perlbase-autoloader +perlbase-autouse +perlbase-dynaloader +perlbase-digest +perlbase-findbin +perlbase-getopt +perlbase-hash
```
Please wrap this line.

wvdakker replied:
Done

pprindeville commented on net/shorewall/files/hotplug_iface:
```diff
@@ -0,0 +1,13 @@
+        # might need to restore some routing
+        /etc/init.d/shorewall restart
+        ;;
+esac
```
Need newline on last line.  Please set your editor's profile to not allow unterminated files.

wvdakker replied:
Done

wvdakker commented:
pprindeville: ping

lucize commented:
@pprindeville can you review this ?

champtar commented:
ping @pprindeville

wvdakker commented:
@champtar : can you merge the request? pprindeville is non-responsive on this PR.
